mewarpx/mewarpx/postprocess_store/movie.py

```diff
@@ -0,0 +1,116 @@
+
+Requires `ffmpeg` software to be installed and accessible from the command
+line.
+"""
+import glob
+import logging
+import os
+import subprocess
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from mewarpx.postprocess_store import pputil
+from mewarpx.utils_store import plotting
+from mewarpx.utils_store import util as mwxutil
+
+logger = logging.getLogger(__name__)
+
+
+def mk_movie(dirname, prefix, valmin, valmax, template, fps=5,
+             subdir='diags/fields', max_num=np.inf, overwrite=False,
+             write_images_only=False, **kwargs):
+    """Create a new movie file.
+
+    This works by first plotting the image at each timestep, stored in
+    ``processed_images`` under the run directory, and then using ``ffmpeg`` to
+    concatenate them into a movie, storied in ``movies``. ``valmin`` and
+    ``valmax`` are required as they ensure the color scale is constant
+    throughout the movie.
+
+    Arguments:
+        dirname (str): Path to the primary simulation run directory.
+        prefix (str): The filename prefix of the diagnostic files to print,
+            e.g. ``Electrostatic_potential`` for
+            ``Electrostatic_potential_0000001000.npy``-type files.
+        template (str): Used to determine default labels. See
+            :class:`mewarpx.utils_store.plotting.ArrayPlot` for more
+            information.
+        valmin (float): If not None, override default minimum of the array
+            being plotted for color scales etc.
+        valmax (float): If not None, override default maximum of the array
+            being plotted for color scales etc.
+        fps (int): Frames per second for the final movie. Default 5.
+        subdir (str): The subdirectory the diagnostic files are in, default
+            ``diags/fields``.
+        max_num (int): Max number of frames to create, usually to generate a
+            short movie for testing purposes. Default infinity.
+        overwrite (bool): If True, overwrite any already-processed images.
+            Necessary if any plotting parameters have changed. Otherwise False
+            will only create new plots, useful if no parameters changed.
+            Default False.
+        write_images_only (bool): If True, don't call ffmpeg to make a movie,
+            only make the image frames. Useful when ffmpeg not installed.
+            Default False.
+        kwargs: All other kwargs are passed to
+            :class:`mewarpx.utils_store.plotting.ArrayPlot`.
+    """
+    pputil.load_run(dirname)
+    processed_img_dir = os.path.join(dirname, 'processed_images')
+    movies_dir = os.path.join(dirname, 'movies')
+    mwxutil.mkdir_p(processed_img_dir)
+    mwxutil.mkdir_p(movies_dir)
+
+    filename_pattern = prefix + '_{:10d}'
+    numpy_file_pattern = os.path.join(
+        dirname, subdir, filename_pattern + ".npy")
+    glob_file_pattern = numpy_file_pattern.replace(
+        "{:10d}", "[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]"
```

Comment:
I think this should work:
```suggestion
        "{:10d}", "[0-9]"*10
```